exchanges/order/orders.go

```diff
@@ -129,6 +129,23 @@ func (s *Submit) Validate(requirements protocol.TradingRequirements, opt ...vali
 	return nil
 }
 
+// GetTradeAmount returns the trade amount based on the exchange's trading
+// requirements. Some exchanges depending on direction and order type use
+// quotation (funds in balance) or base amounts. If the exchange does not have
+// any specific requirements it will return the base amount.
+func (s *Submit) GetTradeAmount(tr protocol.TradingRequirements) float64 {
```

Comment:
Unless you have experienced this against other exchanges, this function looks like it should be localised to gateIO rather than part of the submit